flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/VectorizedColumnBatch.java

```diff
@@ -132,4 +135,43 @@ public Decimal getDecimal(int rowId, int colId, int precision, int scale) {
 			return Decimal.fromUnscaledBytes(precision, scale, bytes);
 		}
 	}
+
+	public SqlTimestamp getTimestamp(int rowId, int colId, int precision) {
+		if (isNullAt(rowId, colId)) {
+			return null;
+		}
+
+		// The precision of Timestamp in parquet should be one of MILLIS, MICROS or NANOS.
+		// https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#timestamp
+		//
+		// For MILLIS, the underlying INT64 holds milliseconds
+		// For MICROS, the underlying INT64 holds microseconds
+		// For NANOS, the underlying INT96 holds nanoOfDay(8 bytes) and julianDay(4 bytes)
+		if (columns[colId] instanceof TimestampColumnVector) {
```

Comment:
I don't quite understand what's going on here. Looks like we have our own `TimestampColumnVector`, and use this to deal with some situations. But we also have other shortcuts for SqlTimestamp? Did we setup the rule or we follow the rule of parquet, like use a long vector for timestamp with precision less than 3? If we use parquet's protocol, what will happen when we reading from orc?

Reply:
for Parquet, we use LongColumnVector (for int64, the underlying long has different meaning) or BytesColumnVector (for int96).
for Orc, we use a OrcTimestampColumnVector which holds a HIve's TimestmapColumnVector and implement our `TimestampColumnVector` interface.

Reply:
I'm not sure this is right. This class should have Flink's own protocol?

Reply:
OK, I will remove the Parquet protocal part. and the Parquet reader should holds a Long or Bytes vector and implement the `TimestmapColumnVector` interface.